api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/InitialArtStartDateCalculation.java

```diff
@@ -0,0 +1,117 @@
+package org.openmrs.module.eptsreports.reporting.calculation.pvls;
+
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Obs;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.context.Context;
+import org.openmrs.calculation.patient.PatientCalculationContext;
+import org.openmrs.calculation.result.CalculationResultMap;
+import org.openmrs.calculation.result.SimpleResult;
+import org.openmrs.module.eptsreports.reporting.calculation.AbstractPatientCalculation;
+import org.openmrs.module.eptsreports.reporting.calculation.EptsCalculations;
+import org.openmrs.module.eptsreports.reporting.utils.EptsCalculationUtils;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Calculates the date on which a patient first started ART
+ * 
+ * @return a CulculationResultMap
+ */
+public class InitialArtStartDateCalculation extends AbstractPatientCalculation {
+	
+	/**
+	 * @should return null for patients who have not started ART
+	 * @should return start date for patients who have started ART
+	 * @see org.openmrs.calculation.patient.PatientCalculation#evaluate(java.util.Collection,
+	 *      java.util.Map, org.openmrs.calculation.patient.PatientCalculationContext)
+	 */
+	@Override
+	public CalculationResultMap evaluate(Collection<Integer> cohort, Map<String, Object> parameterValues,
+	        PatientCalculationContext context) {
+		
+		// Get calculation map date for the first program enrollment
+		CalculationResultMap map = new CalculationResultMap();
+		// only get patients who are alive
+		ConceptService conceptService = Context.getConceptService();
+		Concept arvPlan = conceptService.getConceptByUuid("e1d9ee10-1d5f-11e0-b929-000c29ad1d07");
+		Concept drugStartDate = conceptService.getConceptByUuid("e1d8f690-1d5f-11e0-b929-000c29ad1d07");
+		Concept startDrugs = conceptService.getConceptByUuid("e1d9ef28-1d5f-11e0-b929-000c29ad1d07");
+		
+		EncounterType encounterTypePharmacy = Context.getEncounterService()
+		        .getEncounterTypeByUuid("e279133c-1d5f-11e0-b929-000c29ad1d07");
+		EncounterType adultoSeguimento = Context.getEncounterService().getEncounterTypeByUuid("e278f956-1d5f-11e0-b929-000c29ad1d07");
+		EncounterType arvPaed = Context.getEncounterService().getEncounterTypeByUuid("e278fce4-1d5f-11e0-b929-000c29ad1d07");
+		
+		CalculationResultMap inProgramMap = calculate(new InHivProgramEnrollmentCalculation(), cohort, context);
```

Comment:
@ningosi seems we are not restricting the location on the queries triggered by these 4 lines

Reply:
The calculation module has only the OnDate paramater and the base cohort that it is run on. This calculation is meant to be intersected by other definitions that caters for location, start date and end date plus other parameters that can be supplied. However, the passed cohort can be overriden by supplying a custom cohort that will pass those parameters.